cansat-gps/tests/main.rs

```diff
@@ -15,29 +15,40 @@ fn gps_last_nmea_returns_none_if_no_msg_received() {
     assert!(gps.last_nmea().is_none());
 }
 
-#[ignore]
 #[test]
 fn gps_last_nmea_returns_msg_if_msg_received() {
     let uart = mock::Serial::new(FIRST_MSG.to_owned());
-    let _ = Gps::new(uart);
+    let mut gps = Gps::new(uart);
+
+    // Read until the first message is fully received (the second element of the tuple returned by read_serial() is true)
+    // So as long as it is 'false', it means that the read operation has not yet been completed, and only when the value becomes 'true', it means that the entire GPS message has been read.
+    while let Ok((_read_data, false)) = gps.read_serial() {}
 
-    todo!()
+    // Check if the message has been successfully received (i.e., when in 'is_some()' the returned value is not empty = true).
+    assert!(gps.last_nmea().is_some());
 }
 
-#[ignore]
 #[test]
 fn gps_last_nmea_returns_first_msg_if_second_not_yet_terminated() {
     let uart = mock::Serial::new([FIRST_MSG, SECOND_MSG].concat());
-    let _ = Gps::new(uart);
+    let mut gps = Gps::new(uart);
 
-    todo!()
+    // Similar to the comments above
+    while let Ok((_read_data, false)) = gps.read_serial() {}
+
```

Comment:
Ok, so the (not clearly communicated) idea here was to start reading the second message before calling `last_nmea()` to ensure that it will return the last complete msg. Just call `read_serial().unwrap()` 2 or 3 times before the assert.